File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/events/SLLoginResultEvent.java
package com.lumiyaviewer.lumiya.slproto.events;

import java.util.UUID;

public class SLLoginResultEvent {
    public final UUID activeAgentUUID;
    public final String message;
    public final boolean success;

    public SLLoginResultEvent(boolean z, String str, UUID uuid) {
        this.success = z;
        this.message = str;
        this.activeAgentUUID = uuid;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/events/SLLoginResultEvent.kt
package com.lumiyaviewer.lumiya.slproto.events

import java.util.UUID

class SLLoginResultEvent private constructor(
    @JvmField val success: Boolean,
    @JvmField val message: String,
    @JvmField val activeAgentUUID: UUID,
    /** The grid asked for a multi-factor code; login can be retried with one. */
    @JvmField val mfaRequired: Boolean
) {
    constructor(success: Boolean, message: String, uuid: UUID) : this(success, message, uuid, false)

    companion object {
        @JvmStatic
        fun mfaChallenge(message: String, uuid: UUID): SLLoginResultEvent {
            return SLLoginResultEvent(false, message, uuid, true)
        }
    }
}
